libraries/kotlin.test/common/src/test/kotlin/kotlin/test/tests/BasicAssertionsTest.kt

```diff
@@ -88,6 +88,119 @@ class BasicAssertionsTest {
         checkFailedAssertion { assertSame(instance1, instance2) }
     }
 
+    @Test
+    fun testAssertEqualsDouble() {
+        assertEquals(0.01, 0.02, .01)
+
+        // 0.0, -0.0
+        assertEquals(0.0, 0.0, 0.0)
+        assertEquals(0.0, -0.0, 0.0)
+        assertEquals(-0.0, 0.0, 0.0)
+        assertEquals(0.0, 0.0, -0.0)
+        assertEquals(0.0, -0.0, -0.0)
+        assertEquals(-0.0, 0.0, -0.0)
+
+        // NaN
+        val nans = doubleArrayOf(Double.NaN, Double.fromBits(0xFFF80000L shl 32), Double.fromBits(Double.NaN.toBits() or 1))
+        for (nan1 in nans) {
+            assertTrue(nan1.isNaN())
+            for (nan2 in nans) {
+                assertEquals(nan1, nan2, 0.1)
+                assertEquals(nan1, nan2, 0.0)
+            }
+        }
+
+        // MIN_VALUE, MAX_VALUE
+        assertEquals(Double.MAX_VALUE, Double.MAX_VALUE, 0.0)
+        assertEquals(Double.MIN_VALUE, Double.MIN_VALUE, 0.0)
+        assertEquals(Double.MAX_VALUE, Double.MIN_VALUE, Double.MAX_VALUE)
+        assertEquals(Double.MIN_VALUE, Double.MAX_VALUE, Double.MAX_VALUE)
+
+        // POSITIVE_INFINITY, NEGATIVE_INFINITY
+        assertEquals(Double.POSITIVE_INFINITY, Double.POSITIVE_INFINITY, 0.0)
+        assertEquals(Double.NEGATIVE_INFINITY, Double.NEGATIVE_INFINITY, 0.0)
+        assertEquals(Double.POSITIVE_INFINITY, Double.NEGATIVE_INFINITY, Double.POSITIVE_INFINITY)
+        assertEquals(Double.NEGATIVE_INFINITY, Double.POSITIVE_INFINITY, Double.POSITIVE_INFINITY)
+    }
+
+    @Test
+    fun testAssertEqualsDoubleFails() {
+        checkFailedAssertion { assertEquals(0.01, 1.03, .01) }
+
+        // negative absoluteTolerance
+        checkFailedAssertion { assertEquals(0.01, 1.03, -5.0) }
+
+    }
+
+    @Test
+    fun testAssertEqualsFloat() {
+        assertEquals(0.01f, 0.02f, .01f)
+
+        // 0.0, -0.0
+        assertEquals(0.0f, 0.0f, 0.0f)
+        assertEquals(0.0f, -0.0f, 0.0f)
+        assertEquals(-0.0f, 0.0f, 0.0f)
+        assertEquals(0.0f, 0.0f, -0.0f)
+        assertEquals(0.0f, -0.0f, -0.0f)
+        assertEquals(-0.0f, 0.0f, -0.0f)
+
+        // NaN
+        val nans = floatArrayOf(Float.NaN, Float.fromBits(0xFFFC0000.toInt()), Float.fromBits(Float.NaN.toBits() or 1))
```

Comment:
Minor: it may be enough to test for positive and negative NaNs.

Reply:
done